### PR TITLE
misc(cli): use scallop instead of sumac

### PR DESCRIPTION
**Pull request by szymonm**
**Pull Request checklist**

- [x] The commit(s) message(s) follows the contribution [guidelines](CONTRIBUTING.md) ?
- [x] Tests for the changes have been added (for bug fixes / features) ?
- [x] Docs have been added / updated (for bug fixes / features) ?

**Current behavior :** 

Part of #457 

`quantifind.sumac` is not being ported to Scala 2.12. Instead, I migrate `CliMain` to use scallop
that has already been used in `gateway`.

**BREAKING CHANGES**

Scallop uses different way of parsing lists and maps.
Specifically, uses space (` `) for list delimiter and equals (`=`) for value definition in map.
This applies to following parameters of `CliMain`:
1.  `labelNames`: instead of `--labelNames a,b,c` program expects `--labelNames a b c`.
2. `shards`: instead of `--shards a,b,c` program expects `--shards a b c`.
3. `labelFilter`: instead of `--labelFilter key1:value1,key2:value2` it expects `--labelFilter key1=value1 key2=value2`.

### CODE REVIEW

**Comment by broneill** [comment]
You might need to rebase from the develop branch for the build to pass.

**Comment by szymonm** [comment]
Fixed, @broneill

**Comment by szymonm** [comment]
Are there any concerns?

Are the breaking changes OK? I can do some trickery to keep backwards compatibility, but that will be less elegant.

Also, I see that some docs around this will need to be changed.

**Comment by broneill** [comment]
I'd like to get some feedback from @velvia .

**Comment by velvia** [comment]
I’ll have a look soon, like today or over the weekend…….

**Comment by szymonm** [comment]
Pinging @velvia

File: cli/src/main/scala/filodb.cli/CliMain.scala [hunk 1]
```diff
@@ -159,45 +161,45 @@ object CliMain extends ArgMain[Arguments] with FilodbClusterNode {
...
         case Some("labelValues") =>
-          require(args.host.nonEmpty && args.dataset.nonEmpty && args.labelNames.nonEmpty, "--host, --dataset and --labelName must be defined")
-          val remote = Client.standaloneClient(system, args.host.get, args.port)
-          val options = QOptions(args.limit, args.sampleLimit, args.everyNSeconds.map(_.toInt),
-            timeout, args.shards.map(_.map(_.toInt)), args.spread)
-          parseLabelValuesQuery(remote, args.labelNames, args.labelFilter, args.dataset.get,
+          require(args.host.isDefined && args.dataset.isDefined && args.labelNames.isDefined, "--host, --dataset and --labelName must be defined")
+          val remote = Client.standaloneClient(system, args.host(), args.port())
+          val options = QOptions(args.limit(), args.sampleLimit(), args.everyNSeconds.map(_.toInt).toOption,
+            timeout, args.shards.map(_.map(_.toInt)).toOption, args.spread.toOption.map(Integer.valueOf))
```
Comment (velvia) [inline cli/src/main/scala/filodb.cli/CliMain.scala:191]: Is `Integer.valueOf` for `args.spread` necessary?  Spread is declared with `opt[Int]`....

Comment (szymonm) [reply cli/src/main/scala/filodb.cli/CliMain.scala:191]: `QOptions` constructor takes `Option[Integer]` (don't know the reason, seems like unfounded inconsistency).

**Pull request merged**